[PE-D][Tester B] Application crash on startup due to unvalidated storage data

## Description:
The application experiences a fatal crash during the startup sequence if the underlying data storage file (`onlytutors.json`) contains inconsistent information—specifically, a time range where the start time is set after the end time. Instead of handling this data inconsistency gracefully, the entire system fails to initialize, preventing the user from accessing the app at all.

## Steps to reproduce:
1. Open `data/onlytutors.json`.
2. Change the `startTime` for `Alex` (The first student) from 11:00 to 12:00.
3. Save the change.
4. Reopen the application using `java -jar "onlytutors.jar"` command.

## Expected:
The application should launch successfully. It should ideally detect the inconsistent data and either:
* Flag the specific record as corrupted while allowing the rest of the app to function.
* Prompt the user that the data file is corrupted and offer to start with a clean state.

## Actual:
The application terminates immediately during the "Initializing" phase. A system error is thrown, and the GUI never appears, rendering the software completely unusable until the data file is manually corrected or deleted.

<img width="1312" height="912" alt="Image" src="https://raw.githubusercontent.com/NUS-CS2103-AY2526-S2/ped/master/files/4dc376fe-6545-408b-8d5a-420c5c77ca9a.png" />

<img width="972" height="671" alt="Image" src="https://raw.githubusercontent.com/NUS-CS2103-AY2526-S2/ped/master/files/c4fbc918-83df-4cff-bf4c-35e28dc8f352.png" />

> The first image shows the `onlytutors.json` file after change, and the second image shows the error message on the terminal when trying to open the application again.

## The error message:
Apr 10, 2026 4:19:38 PM seedu.address.Main main
WARNING: The warning about Unsupported JavaFX configuration below (if any) can be ignored.
Apr 10, 2026 4:19:38 PM seedu.address.MainApp init
INFO: =============================[ Initializing AddressBook ]===========================
Apr 10, 2026 4:19:38 PM seedu.address.MainApp initConfig
INFO: Using config file : config.json
Apr 10, 2026 4:19:38 PM seedu.address.commons.util.JsonUtil readJsonFile
INFO: JSON file config.json found.
Apr 10, 2026 4:19:38 PM seedu.address.commons.core.LogsCenter init
INFO: Log level will be set as: INFO
Apr 10, 2026 4:19:38 PM seedu.address.MainApp initPrefs
INFO: Using preference file : preferences.json
Apr 10, 2026 4:19:38 PM seedu.address.commons.util.JsonUtil readJsonFile
INFO: JSON file preferences.json found.
Apr 10, 2026 4:19:38 PM seedu.address.MainApp initModelManager
INFO: Using data file : data/onlytutors.json
Apr 10, 2026 4:19:38 PM seedu.address.commons.util.JsonUtil readJsonFile
INFO: JSON file data/onlytutors.json found.
Exception in Application init method
Exception in thread "main" java.lang.RuntimeException: Exception in Application init method
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:888)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.IllegalArgumentException: The end time must be strictly after the start time.
	at seedu.address.model.person.Person.<init>(Person.java:42)
	at seedu.address.storage.JsonAdaptedPerson.toModelType(JsonAdaptedPerson.java:159)
	at seedu.address.storage.JsonSerializableAddressBook.toModelType(JsonSerializableAddressBook.java:51)
	at seedu.address.storage.JsonAddressBookStorage.readAddressBook(JsonAddressBookStorage.java:55)
	at seedu.address.storage.StorageManager.readAddressBook(StorageManager.java:64)
	at seedu.address.storage.StorageManager.readAddressBook(StorageManager.java:58)
	at seedu.address.MainApp.initModelManager(MainApp.java:81)
	at seedu.address.MainApp.init(MainApp.java:63)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:817)
	... 2 more

## Suggestion:
I think this bug is nearly impossible to avoid, but you may try to solve it when relaunching the app by either flag the specific record as corrupted while allowing the rest of the app to function, or prompt the user that the data file is corrupted and offer to start with a clean state. Or I think maybe you can also fix this bug by not allowing users to access and modify the data file.

-------------
Labels: `severity.High` `type.FeatureFlaw`

--- a/docs/DeveloperGuide.md
+++ b/docs/DeveloperGuide.md
@@ -650,14 +650,14 @@ testers are expected to do more *exploratory* testing.
 
 1. Dealing with missing data file
 
-   1. Delete the `data/onlytutors.json` file if it exists.
+   1. Delete the `data/addressbook.json` file if it exists.
 
    1. Launch the app.<br>
       Expected: App starts with a set of sample student data.
 
 1. Dealing with corrupted data file
 
-   1. Open `data/onlytutors.json` and introduce invalid content (e.g., delete a required field or add invalid characters).
+   1. Open `data/addressbook.json` and introduce invalid content (e.g., delete a required field or add invalid characters).
 
    1. Launch the app.<br>
       Expected: App starts with an empty student list. A warning may be logged.

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -622,7 +622,7 @@ OnlyTutors data is saved to the hard disk automatically after any command that c
 
 ### Editing the data file
 
-OnlyTutors data is saved automatically as a JSON file at `[JAR file location]/data/onlytutors.json`. Advanced users are welcome to update data directly by editing that file.
+OnlyTutors data is saved automatically as a JSON file at `[JAR file location]/data/addressbook.json`. Advanced users are welcome to update data directly by editing that file.
 
 <div markdown="span" class="alert alert-warning">:exclamation: **Warning:**
 If your changes to the data file make its format invalid, OnlyTutors may discard all data and start with an empty data file at the next run. It is recommended to take a backup of the file before editing it. Furthermore, certain edits can cause OnlyTutors to behave in unexpected ways (e.g., if a value entered is outside the acceptable range). Edit the data file only if you are confident that you can update it correctly.

--- a/src/main/java/seedu/address/model/UserPrefs.java
+++ b/src/main/java/seedu/address/model/UserPrefs.java
@@ -14,7 +14,7 @@ import seedu.address.commons.core.GuiSettings;
 public class UserPrefs implements ReadOnlyUserPrefs {
 
     private GuiSettings guiSettings = new GuiSettings();
-    private Path addressBookFilePath = Paths.get("data" , "onlytutors.json");
+    private Path addressBookFilePath = Paths.get("data" , "addressbook.json");
 
     /**
      * Creates a {@code UserPrefs} with default values.

--- a/src/main/java/seedu/address/storage/JsonAdaptedPerson.java
+++ b/src/main/java/seedu/address/storage/JsonAdaptedPerson.java
@@ -147,6 +147,10 @@ class JsonAdaptedPerson {
         }
         final Time modelEndTime = new Time(endTime);
 
+        if (!modelEndTime.isAfter(modelStartTime)) {
+            throw new IllegalValueException(Time.MESSAGE_COMPARISON_CONSTRAINTS);
+        }
+
         if (rate == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Rate.class.getSimpleName()));
         }

--- a/src/test/data/JsonUserPrefsStorageTest/ExtraValuesUserPref.json
+++ b/src/test/data/JsonUserPrefsStorageTest/ExtraValuesUserPref.json
@@ -9,5 +9,5 @@
       "z" : 99
     }
   },
-  "addressBookFilePath" : "onlytutors.json"
+  "addressBookFilePath" : "addressbook.json"
 }

--- a/src/test/data/JsonUserPrefsStorageTest/TypicalUserPref.json
+++ b/src/test/data/JsonUserPrefsStorageTest/TypicalUserPref.json
@@ -7,5 +7,5 @@
       "y" : 100
     }
   },
-  "addressBookFilePath" : "onlytutors.json"
+  "addressBookFilePath" : "addressbook.json"
 }

--- a/src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java
+++ b/src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java
@@ -182,6 +182,18 @@ public class JsonAdaptedPersonTest {
     }
 
     @Test
+    public void toModelType_startTimeAfterEndTime_throwsIllegalValueException() {
+        String startTime = "18:00";
+        String endTime = "08:00";
+        JsonAdaptedPerson person =
+                new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS,
+                        VALID_DAY, startTime, endTime, VALID_RATE, false, VALID_TAGS);
+
+        String expectedMessage = Time.MESSAGE_COMPARISON_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, person::toModelType);
+    }
+
+    @Test
     public void toModelType_invalidRate_throwsIllegalValueException() {
         JsonAdaptedPerson person =
                 new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS,

--- a/src/test/java/seedu/address/storage/JsonUserPrefsStorageTest.java
+++ b/src/test/java/seedu/address/storage/JsonUserPrefsStorageTest.java
@@ -73,7 +73,7 @@ public class JsonUserPrefsStorageTest {
     private UserPrefs getTypicalUserPrefs() {
         UserPrefs userPrefs = new UserPrefs();
         userPrefs.setGuiSettings(new GuiSettings(1000, 500, 300, 100));
-        userPrefs.setAddressBookFilePath(Paths.get("onlytutors.json"));
+        userPrefs.setAddressBookFilePath(Paths.get("addressbook.json"));
         return userPrefs;
     }
 

